sdk/cosmosdb/cosmos/test/public/functional/query.spec.ts

```diff
@@ -109,6 +110,42 @@ describe("Queries", function (this: Suite) {
       assert(counter === 2, "iterator should have run 3 times");
     });
 
+    describe("asyncIterator with operationOptions", function (this: Suite) {
+      it("RU cap not breached", async function () {
+        const queryIterator = resources.container.items.readAll({ maxItemCount: 2 });
+        let counter = 0;
+        for await (const { resources: docs } of queryIterator.getAsyncIterator({
+          ruCapPerOperation: 20,
+        })) {
+          if (counter === 0) {
+            assert.equal(docs[0].id, resources.doc1.id, "first document should be doc1");
+            assert.equal(docs[1].id, resources.doc2.id, "second document should be doc2");
+          } else {
+            assert.equal(docs[0].id, resources.doc3.id, "third document should be doc3");
+          }
+          counter++;
+        }
+        assert(counter === 2, "iterator should have run 3 times");
+      });
+
+      it("RU Cap breached", async function () {
+        const queryIterator = resources.container.items.readAll({ maxItemCount: 2 });
+        try {
+          for await (const { resources: docs } of queryIterator.getAsyncIterator({
+            ruCapPerOperation: 2,
```

Comment:
can we calculate RU cap dynamically here. Since it would be diferent for Routing, Compute, emulator etc. And target a tighter test. for example instead of having 20 and 2, in this and previous test, It should be `calculated_ru_threshold + offset`. Here offset should be as small as possible.